src/store/configureStore.js

```diff
@@ -114,9 +120,39 @@ const middlewares = applyMiddleware(
 
 let store = null
 
+// As long as the mobile client is dependent on the web client, we need to sync
+// the client store from web -> mobile
+const clientStoreKeys = Object.keys(clientStoreReducers)
+
+const syncClientStateToNativeMobile = store => {
+  if (NATIVE_MOBILE) {
+    let currentState
+    const postMessageDebounced = debounce(postMessage, 500, { leading: true })
+
+    store.subscribe(() => {
+      const state = store.getState()
+      const previousState = currentState
+      currentState = state
+      if (
+        !previousState ||
+        clientStoreKeys.some(k => currentState[k] !== previousState[k])
+      ) {
+        // Debounce messages to minimize expensive stringify and parse.
```

Comment:
This debounce can be a lot smaller.

console.time('t');a = JSON.stringify(store.getState()); JSON.parse(a);console.timeEnd('t')

If you run this from the client in the console it's < 5ms. i believe mobile environments should be quite similar. not entirely sure what the overhead of the message pass is, but it's still not going to be 500ms